### ISSUE TITLE
Sharepoint Integration

### ISSUE THREAD

**Issue by MarsRomer** [bcgov/CONN-CCBC-portal#1417]
As an analyst
I want to click on file names in the portal that exist in sharepoint
So that they can open for me without my need for looking for it in sharepoint

**Description:**
This ticket is to explore the idea of integration sharepoint into our portal.  
What are the pro's/con's, risks, and cost/benefit of doing this?

# Acceptance Criteria:

## {Page Name}

**Screen**

[wireframe 1]
[wireframe 2]

- [ ] .

**Page Behaviours**

- [ ] .
- [ ] .
- [ ] .

**Validations**

- [ ] .
- [ ] .
- [ ] .

**Actions**

- [ ] .
- [ ] .
- [ ] .

### Not included:
- .
- .

## Implementation Details
- **DB:**
  - .
  - .
- **Middleware:**
  - .
  - .
- **Frontend:**
  - .
  - .
- **Tests:**
  - .
  - .
- **Documentation:**
  - .
  - .

**Comment by elliottSlee** [comment]
UX/UI considerations:
- Uploading a file simultaneously uploads to s3 bucket & Sharepoint
- Virus scan occurs before putting on Sharepoint, perhaps re-use empty file for flagged files
- If it's a zip file, extract it then put contents on Sharepoint
- Files are organized according to existing file structure
- Nice to have: Click a file in the portal opens file on Sharepoint/Microsoft 365
- Nice to have: If file added to Sharepoint, UI to assign it to correct place in Portal

**Comment by rafasdc** [comment]
Attempting to authenticate using email/pass combination returns error code:

> AADSTS53003: AADSTS53003: Access has been blocked by Conditional Access policies. The access policy does not allow token issuance. 

We will likely have to use one of the two methods mentioned here, depending which is recommended and/or supported on the BCGov tenancy: https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly?source=recommendations
